lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -263,6 +279,35 @@ def generate_crontab_if_necessary(*args, **kwargs) -> int:
 
         assert ret_val == 43
 
+    @staticmethod
+    @pytest.mark.parametrize("init_indexing_exc", ["section", "option"])
+    def test_main_init_indexing_failed(
+        monkeypatch, make_logger, mock_get_server_config, init_indexing_exc
+    ):
+        def noop(*args, **kwargs):
+            pass
+
+        called = [False]
+
+        def init_indexing(*args, **kwargs) -> int:
+            called[0] = True
+            if init_indexing_exc == "section":
+                exc = NoSectionError("missingsection")
+            elif init_indexing_exc == "option":
+                exc = NoOptionError("section", "missingoption")
+            else:
+                exc = Exception(f"Bad test parameter, {init_indexing_exc}")
```

Comment:
Rather than having the `if`/`else` here (which leads to the odd case at 299, which I don't think really means what you think it means...), how about having `init_indexing_exc` actually _be_ the exception object that you want raised?  (It's trivial to move the contents of lines 295 and 297 to the `parametrize()` declaration at line 283.)  And, then you don't need the third case here.

Reply:
Fixed.